### PR TITLE
chore: bump rust version to latest nightly

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**

### CODE REVIEW

File: pallets/gear-debug/src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::manual_inspect)]
```
Comment (NikVolf) [inline pallets/gear-debug/src/lib.rs:20]: add an issue to remove this?

File: utils/gring/src/keystore.rs [hunk 2]
```diff
@@ -137,6 +137,7 @@ impl Keystore {
     }
 
     /// Returns self with the given name in meta.
+    #[allow(clippy::assigning_clones)]
```
Comment (NikVolf) [inline utils/gring/src/keystore.rs:140]: here also -- add an issue

**Review by NikVolf — APPROVED** [review]

File: pallets/gear-debug/src/lib.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply pallets/gear-debug/src/lib.rs:20]: I'll create, maybe we need to update substrate

Comment (StackOverflowExcept1on) [reply pallets/gear-debug/src/lib.rs:20]: #4097

**Review by ukint-vs — APPROVED** [review]

**Pull request merged**